### PR TITLE
R/jdbc read

### PR DESCRIPTION
**Pull request by kovrus**

### CODE REVIEW

File: pages/docs/tools/streamsets.html [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+database. To keep the manual short, we would refer to the StreamSets tutorial
+explaining how to [configure the directory origin][6] for reading
+[CSV sample data][7], which would be further used as the origin stage in the
+pipeline with the Crate JDBC destination.
+
+## Crate JDBC Destination
+
+The Crate JDBC Producer destination uses the JDBC connection to write data to
+the database. In this section, we show how to build the StreamSets project for
+injecting CSV records from the local FS into the Crate database with the
```
Comment (chaudum) [inline pages/docs/tools/streamsets.html]: filesystem

File: pages/docs/tools/streamsets.html [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+preliminary preprocessing of certain records fields.
+
+1. Create the [new pipeline][8].
+![Create the new pipeline]({% media '/media/1605/1-streamsets-dest-new.png' %})
+
+2. Follow the above mentioned StreamSets [tutorial][6] which explains how to
+create the directory origin and read CSV files from local FS. Use the data
```
Comment (chaudum) [inline pages/docs/tools/streamsets.html]: filesystem

File: pages/docs/tools/streamsets.html [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+5. The next step is to configure the Crate JDBC destination. First, the driver
+must be loaded and an optional health check query can be provided.
+![Load the Crate JDBC driver]({% media '/media/1605/3-streamsets-dest-driver.png' %})
+Then, we configure the JDBC connection (1), define the destination table (2)
+and map table columns to input fields (3). The mapping of columns can be
+omitted if the output fields from the origin or processing stage are identical
+to input fields of the destination stage. After the above-mentioned
+configurations steps we still can see warnings in the *Credential tab*. We
+solved it by unchecking the user authorization box. Such as Crate does not
```
Comment (chaudum) [inline pages/docs/tools/streamsets.html]: .... authorization box, because Crate ...

File: pages/docs/tools/streamsets.html [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+The JDBC Consumer origin uses the JDBC connection to read data from the Crate
+database. The Crate origin stage returns data as a map with column names and
+field values.
+
+Currently, the usage of the Crate JDBC driver in combination with StreamSets
+Data Collector introduces few limitations:
+
+- Only incremental mode for fetching data is supported.
+- The offset column **MUST** be the primary key column.
+- The **LIMIT** clause must be explicitly set in the query.
```
Comment (chaudum) [inline pages/docs/tools/streamsets.html]: better: \`LIMIT\`

File: pages/docs/tools/streamsets.html [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+- Batch size in the driver configuration is ignored.
+
+It is important to mention that you have to adjust the LIMIT size in the query
```
Comment (chaudum) [inline pages/docs/tools/streamsets.html]: \`LIMIT\`

File: pages/docs/tools/streamsets.html [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+in the StreamSets.
+
+You can download the latest standalone version directly from the
+[Bintray Repository][4].
+
+After the JDBC driver is downloaded to the arbitrary destination, you can
+proceed with the installation of the driver. We strictly recommend following
+the StreamSets instruction for installing [additional drivers][5].
+
+The next two sections provide a brief introduction into building simple data
```
Comment (chaudum) [inline pages/docs/tools/streamsets.html]: the first 2 sentences do basically say the same thing. try to combine them

File: pages/docs/tools/streamsets.html [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+stream pipelines using Crate with StreamSets Data Collector. As an example,
+we demonstrate projects showing how to read data from and write to the Crate
+database. To keep the manual short, we would refer to the StreamSets tutorial
```
Comment (chaudum) [inline pages/docs/tools/streamsets.html]: try to combine the 2 sentences

File: pages/docs/tools/streamsets.html [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+directory origin does not perform any preliminary preprocessing of records.
+All source fields are represented as strings in the CSV. However, some of the
+selected fields should be processed and converted into float values.
+Therefore, we add an additional processing [Field Converter][9] stage into the
+pipeline.
+![Convert string input values]({% media '/media/1605/2-streamsets-dst-field-conversion.png' %})
+
+3. Create the taxi table in Crate.
+
+    ```bash
```
Comment (chaudum) [inline pages/docs/tools/streamsets.html]: no indentation

File: pages/docs/tools/streamsets.html [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+## Crate JDBC Origin
+
+The JDBC Consumer origin uses the JDBC connection to read data from the Crate
+database. The Crate origin stage returns data as a map with column names and
+field values.
+
+Currently, the usage of the Crate JDBC driver in combination with StreamSets
+Data Collector introduces few limitations:
+
+- Only incremental mode for the data fetch is supported
```
Comment (chaudum) [inline pages/docs/tools/streamsets.html]: end sentences with `.` or start with lower case

File: pages/docs/tools/streamsets.html [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+data import queries hosted in the [Crate demo repository][14]. Create the
+[rankings table][19] and import the [rankings data][20]. In the demo, we use the
+dataset which contains 18 millions of records. Assuming that the Crate cluster
+is set up and the rankings sample data is imported we can start building the
+data stream pipeline for streaming the data from the Crate database to CSV
+files.
+
+1. Create the [new pipeline][8] and configure the Crate JDBC Driver loading as
+it was done for the JDBC destination configuration. In the JDBC tab of the
+Crate JDBC origin, we set the connection string (1), the SQL query for fetching
```
Comment (chaudum) [inline pages/docs/tools/streamsets.html]: no `,`

File: pages/docs/tools/streamsets.html [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+the data from the database (2), the offset value and column (3). For more
+detail information on how to set the offset value and column see the StreamSets
+[documentation][15].
+![Configure the Crate JDBC driver]({% media '/media/1605/3-streamsets-source-jdbc.png' %})
+
+2. We stream the records from crate to CSV files. In order to accomplish that
```
Comment (chaudum) [inline pages/docs/tools/streamsets.html]: Crate

**Comment by kovrus** [comment]
acked by haudi

**Pull request merged**